src/assets/Azure.Core.Shared/DiagnosticScope.cs

```diff
@@ -310,6 +291,25 @@ public void AddLink(string traceparent, string tracestate, IDictionary<string, s
                         return null;
                     }
 
+                    switch (_kind)
+                    {
+                        case ActivityKind.Internal:
+                            AddTag("kind", "internal");
+                            break;
+                        case ActivityKind.Server:
+                            AddTag("kind", "server");
+                            break;
+                        case ActivityKind.Client:
+                            AddTag("kind", "client");
+                            break;
+                        case ActivityKind.Producer:
+                            AddTag("kind", "producer");
+                            break;
+                        case ActivityKind.Consumer:
+                            AddTag("kind", "consumer");
+                            break;
+                    }
```

Comment:
FYI this change comes from build failure:

> D:\a\_work\1\s\autorest.csharp\eng\SharedCodeCheck.ps1 : Shared source files are updated. Please run eng/DownloadSharedSource.ps1

So I did run `eng/DownloadSharedSource.ps1` locally and pushed the resulting change.